docs/adr/0010-migrate-to-v3-cluster-resource.md

```diff
@@ -0,0 +1,66 @@
+The goal of this ADR is to describe the procedure to migrate to the new controller, which will become the default cluster import mechanism, and, eventually, deprecate the old logic. As users who have already provisioned CAPI clusters will already have v1 cluster resources associated with CAPI clusters, we must ensure that we make a safe transition to the new import mechanism, avoiding resource duplicates or deletions that may cause problems to end users.
+
+## Decision
+
+An initial assessment of the requirements for the migration suggested that we should define a procedure based on deleting existing Rancher cluster resources, migrating to new controller, and letting this new controller take care of creating new Rancher cluster resources for imported clusters that align with the pre-migration configuration.
+
+After some investigation, we realized that this approach can end up causing a Rancher cluster resource duplication. While testing this proof-of-concept, it is important to note the differences between the Rancher cluster resources created when importing a cluster to Rancher:
+- Both v1 and v3 clusters are created to represent every Rancher clusters.
+- In the case of the current controller, we generate a v1 and let Rancher take care of v3.
+- When migrating to the newer controller, we generate a v3 and let Rancher take care of v1.
+
+#### So, what is the difference between both controllers?
+
+The main difference in terms of logic is that, while watching v1 resources, we can use the name of the clusters to identify whether there is an existing resource that represents the CAPI-provisioned cluster.
+
+This means that the resource itself is easily linked to a CAPI cluster via name.
+
+Alternatively, when watching v3 clusters in the new controller, the name of the resource is a semi-random string automatically generated by Kubernetes, which forces us to find a different approach to find the resource of interest. Additionally, v3 clusters are not namespaced but global.
+To simplify the watcher on v3 clusters, we add two labels to the resource to reference the `name` and `namespace` of the CAPI cluster it is linked to.
+
+#### And, what is the difference between v3 clusters for each controller?
+
+Now that we now how we link Rancher cluster -> CAPI cluster, we know that the difference between v3 clusters in both controllers is the existence of the CAPI cluster `name` and `namespace` labels.
+
+This means that, if the new controller we'll be migrating to is able to identify v3 clusters with the required labels, we can simply switch controllers and expect everything to be unchanged and Rancher clusters to maintain the same configuration.
```

Comment:
What problems does this approach have? What if a provisioning cluster does not get management cluster created? How the management cluster created by provisioning differs from the one created by us? It seems quite breaking IMO

Reply:
I would say you cannot import a cluster into Rancher with just the provisioning cluster. If there's a provisioning but not a management cluster resource, looks like the original cluster would not be properly imported. What we're saying is that the differences between the mgmt cluster created by provisioning and created by controller (if there are any) are not relevant to the import mechanism, as long as the controller is able to watch and identify the resources (which require the labels and, additionally, a proper display name).

Reply:
It is likely that if the rancher version is below 2.8, the management cluster for provisioning might not be created by default. Turtles can be installed there, still, using regular installation path described in the [docs](https://turtles.docs.rancher.com/getting-started/install-rancher-turtles/using_rancher_dashboard).